### PR TITLE
Freeze Functionbeat docs at 8.15

### PR DESCRIPTION
**Pull request by kilfoyle**
Functionbeat is deprecated in favour of Elastic Serverless Forwarder. The docs are marked as deprecated, so now we just need to freeze the builds at 8.11 until the end of time.

CI checks should pass once this one is merged: https://github.com/elastic/kibana/pull/172909

Rel: https://github.com/elastic/ingest-docs/issues/608
Rel: https://github.com/elastic/docs/pull/2814 (closed in favour of this PR)

### CODE REVIEW

File: conf.yaml [hunk 1]
```diff
@@ -2278,8 +2278,7 @@ contents:
             prefix:     en/beats/functionbeat
             current:    *stackcurrent
```
Comment (bmorelli25) [inline conf.yaml:2279]: `*stackcurrent` expects 8.12 to be built.
```suggestion
            current:    8.11
```

**Comment by nkammah** [comment]
run docs-test

**Comment by nimarezainia** [comment]
@kilfoyle @bmorelli25   is this issue still relevant?

**Comment by kilfoyle** [comment]
@nimarezainia I think it's still relevant. Since Functionbeat isn't being updated we should freeze the docs (rather than keep building new versions of the Functionbeat guide), but I can't do that until the Kibana code is updated (via https://github.com/elastic/kibana/pull/172909) to remove links from the UI into the docs.

**Comment by nimarezainia** [comment]
> @nimarezainia I think it's still relevant. Since Functionbeat isn't being updated we should freeze the docs (rather than keep building new versions of the Functionbeat guide), but I can't do that until the Kibana code is updated (via [elastic/kibana#172909](https://github.com/elastic/kibana/pull/172909)) to remove links from the UI into the docs.

@kilfoyle its now merged. thank you.

**Comment by kilfoyle** [comment]
@elasticmachine run docs-build rebuild

**Comment by kilfoyle** [comment]
I've opened this PR in the Kibana docs to try and resolve the following failing docs build CI check: https://github.com/elastic/kibana/pull/192491



```
2024-09-10 10:28:42 EDT | INFO:build_docs:Bad cross-document links:
-- | --
  | 2024-09-10 10:28:42 EDT | INFO:build_docs:  Kibana [master]: packages/kbn-doc-links/src/get_doc_links.ts contains broken links to:
  | 2024-09-10 10:28:42 EDT | INFO:build_docs:   - en/beats/functionbeat/master
  | 2024-09-10 10:29:31 EDT | 🚨 Error: The command exited with status 255


```

**Comment by kilfoyle** [comment]
Unfortunately my attempted fix didn't work:

```
2024-09-10 11:50:14 EDT | proc [tsc] Found 1 error.
-- | --
  | 2024-09-10 11:50:14 EDT | proc [tsc]
  | 2024-09-10 11:50:14 EDT | ERROR Error: [tsc] exited with code 2
  | 2024-09-10 11:50:14 EDT | at createFailError (dev_cli_errors.ts:28:24)
  | 2024-09-10 11:50:14 EDT | at proc.ts:109:32
  | 2024-09-10 11:50:14 EDT | at /opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/map.ts:58:33
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber._this._next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/OperatorSubscriber.ts:70:13)
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber.Subscriber.next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/Subscriber.ts:75:12)
  | 2024-09-10 11:50:14 EDT | at /opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/take.ts:60:26
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber._this._next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/OperatorSubscriber.ts:70:13)
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber.Subscriber.next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/Subscriber.ts:75:12)
  | 2024-09-10 11:50:14 EDT | at /opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/filter.ts:72:109
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber._this._next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/operators/OperatorSubscriber.ts:70:13)
  | 2024-09-10 11:50:14 EDT | at OperatorSubscriber.Subscriber.next (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/Subscriber.ts:75:12)
  | 2024-09-10 11:50:14 EDT | at ChildProcess.handler (/opt/buildkite-agent/builds/bk-agent-prod-gcp-1725983089491321816/elastic/kibana-pull-request/kibana/node_modules/rxjs/src/internal/observable/fromEvent.ts:292:52)
  | 2024-09-10 11:50:14 EDT | at ChildProcess.emit (node:events:531:35)
  | 2024-09-10 11:50:14 EDT | at Process.ChildProcess._handle.onexit (node:internal/child_process:294:12)
  | 2024-09-10 11:50:14 EDT | ERROR Unable to build TS project refs
  | 2024-09-10 11:50:15 EDT | 🚨 Error: The command exited with status 1

```

**Comment by kilfoyle** [comment]
Unfortunately my [attempt](https://github.com/elastic/kibana/pull/192491) to fix this failing CI check failed.

**Comment by kilfoyle** [comment]
@elastic/docs-build-guild  Would any of you be familiar with link checking specifically for the Kibana UI links?  The docs-build-pr CI gives me the following error, but I have no idea how to resolve it. Scrubbing the `functionbeat` entry from `get_doc_links.ts` doesn't do the trick.

```
2024-09-10 10:28:42 EDT | INFO:build_docs:Bad cross-document links:
-- | --
  | 2024-09-10 10:28:42 EDT | INFO:build_docs:  Kibana [master]: packages/kbn-doc-links/src/get_doc_links.ts contains broken links to:
  | 2024-09-10 10:28:42 EDT | INFO:build_docs:   - en/beats/functionbeat/master
  | 2024-09-10 10:29:31 EDT | 🚨 Error: The command exited with status 255
  | 2024-09-10 10:29:31 EDT | user command error: exit status 255

```

**Comment by dedemorton** [comment]
>Scrubbing the functionbeat entry from get_doc_links.ts

What did you do to scrubing the entry? I still see functionbeat referenced in main: https://github.com/elastic/kibana/blob/main/packages/kbn-doc-links/src/get_doc_links.ts#L284

Assuming the code no longer references the link, then I think you just need to remove the functionbeat entries in get_doc_links.ts and probably types.ts. 

The docs about the link service are here: https://docs.elastic.dev/docs/kibana-doc-links (kinda minimal, but might help you)

**Comment by dedemorton** [comment]
Implied in my comment: If the code does still include a link to the functionbeat docs, it'll need to be updated (or the link removed completely). You can do it yourself, but it might be easier to get a Kibana developer to do it.

**Comment by kilfoyle** [comment]
I've opened a [Kibana issue](https://github.com/elastic/kibana/issues/193030) to see if we can get Functionbeat scrubbed from the `packages/kbn-doc-links/src/get_doc_links.ts` file.

**Comment by kilfoyle** [comment]
I've opened a [new issue](https://github.com/elastic/kibana/issues/193030) specifically for the Integrations Team, since one remaining link appears to come from a Cloud Watch tutorial that uses Functionbeat.

**Comment by kilfoyle** [comment]
@elasticmachine run docs-build

**Comment by kilfoyle** [comment]
Here's one I haven't seen before:

```
2024-11-05 12:50:57 EST | INFO:build_docs:Error executing: node template/cli.js --template resources/web/template.html --source /tmp/docsbuild/target_repo/raw/en/beats/functionbeat/8.15 --dest /tmp/docsbuild/target_repo/html/en/beats/functionbeat/8.15
-- | --
  | 2024-11-05 12:50:57 EST | INFO:build_docs:---out---
  | 2024-11-05 12:50:57 EST | INFO:build_docs:
  | 2024-11-05 12:50:57 EST | INFO:build_docs:---err---
  | 2024-11-05 12:50:57 EST | INFO:build_docs:unhandled rejection Error: EEXIST, file already exists /tmp/docsbuild/target_repo/html/en/beats/functionbeat/8.15/images/coordinate-map.png
  | 2024-11-05 12:50:57 EST | INFO:build_docs:    at fsError (/node_modules/recursive-copy/lib/copy.js:392:14)
  | 2024-11-05 12:50:57 EST | INFO:build_docs:    at /node_modules/recursive-copy/lib/copy.js:268:11 {
  | 2024-11-05 12:50:57 EST | INFO:build_docs:  errno: 47,
  | 2024-11-05 12:50:57 EST | INFO:build_docs:  code: 'EEXIST',
  | 2024-11-05 12:50:57 EST | INFO:build_docs:  path: '/tmp/docsbuild/target_repo/html/en/beats/functionbeat/8.15/images/coordinate-map.png'
  | 2024-11-05 12:50:57 EST | INFO:build_docs:}
  | 2024-11-05 12:50:57 EST | INFO:build_docs:
  | 2024-11-05 12:50:57 EST | INFO:build_docs:---------

```

**Comment by kilfoyle** [comment]
@elasticmachine run docs-build

**Comment by AndersonQ** [comment]
@kilfoyle,

wild guess here, but could this error be because it's building twice for 8.15?
 - once for the current branch https://github.com/elastic/docs/pull/2895/files#diff-9925a85f7cc77717016f5dc21db102f95ff353476bf96849dff5d65cb94ef02dR2280
 - another for the other branches list: https://github.com/elastic/docs/pull/2895/files#diff-9925a85f7cc77717016f5dc21db102f95ff353476bf96849dff5d65cb94ef02dR2282

perhaps `*stackcurrent` is handled in a special way to avoid this conflict, but a regular branch isn't.

Therefore it might be worth to remove `8.15` from the `branches` list

**Comment by kilfoyle** [comment]
@AndersonQ  Just saw your comment now. I'll give that a try.

**Comment by kilfoyle** [comment]
I think 8.15 might have appeared in the `branches` list twice, which would cause problems for sure. For this test I've tried removing all of these: `{main: master}, 8.x, 8.16, 8.15` and freezing the build at `8.14`.

**Comment by kilfoyle** [comment]
```
2024-11-05 16:48:04 EST | INFO:build_docs:Bad cross-document links:
-- | --
  | 2024-11-05 16:48:04 EST | INFO:build_docs:  Kibana [8.16]: packages/kbn-doc-links/src/get_doc_links.ts contains broken links to:
  | 2024-11-05 16:48:04 EST | INFO:build_docs:   - en/beats/functionbeat/8.16
  | 2024-11-05 16:48:04 EST | INFO:build_docs:  Kibana [8.x]: packages/kbn-doc-links/src/get_doc_links.ts contains broken links to:
  | 2024-11-05 16:48:04 EST | INFO:build_docs:   - en/beats/functionbeat/8.x


```

**Comment by karenzone** [comment]
^^ Looks like we still have a couple of instances in Kibana link service.

**Review by karenzone — APPROVED** [review]
LGTM!  🚀

**Comment by kilfoyle** [comment]
Screenshot 2024-11-06 at 2 00 04 PM

---

Screenshot 2024-11-06 at 2 00 40 PM

**Review by lcawl — APPROVED** [review]
Congrats, lgtm!

**Pull request merged**